conda_recipe_manager/fetcher/artifact_fetcher.py

```diff
@@ -1,33 +1,100 @@
 """
-:Description: TODO
+:Description: Module that provides general Artifact Fetching utilities and factory methods.
 """
 
 from __future__ import annotations
 
-from typing import Final, cast
+from typing import cast
 
 from conda_recipe_manager.fetcher.base_artifact_fetcher import BaseArtifactFetcher
+from conda_recipe_manager.fetcher.exceptions import FetchUnsupportedError
+from conda_recipe_manager.fetcher.git_artifact_fetcher import GitArtifactFetcher
+from conda_recipe_manager.fetcher.http_artifact_fetcher import HttpArtifactFetcher
 from conda_recipe_manager.parser.recipe_reader import RecipeReader
+from conda_recipe_manager.parser.types import SchemaVersion
 from conda_recipe_manager.types import Primitives
+from conda_recipe_manager.utils.typing import optional_str
 
-# Identifying string used to flag temp files and directories created by this module.
-_ARTIFACT_FETCHER_FILE_ID: Final[str] = "crm_artifact_fetcher"
 
+def _render_git_key(recipe: RecipeReader, key: str) -> str:
+    """
+    Given the V0 name for a target key used in git-backed recipe sources, return the equivalent key for the recipe
+    format.
+
+    :param recipe: Parser instance for the target recipe
+    :param key: V0 Name for the target git source key
+    :returns: The equivalent key for the recipe's schema.
+    """
+    match recipe.get_schema_version():
+        case SchemaVersion.V0:
+            return key
+        case SchemaVersion.V1:
+            match key:
+                case "git_url":
+                    return "git"
+                case "git_branch":
+                    return "branch"
+                case "git_tag":
+                    return "tag"
+                case "git_rev":
+                    return "rev"
+                case _:  # If this case happens, a developer made a typo.
+                    return ""
```

Comment:
I've gone back and forth on this line a few times. I think my current thinking is that the risk of this silently failing is extremely low and should be caught in any sufficient unit test coverage.

Reply:
I've changed my mind. It is within reason that this could throw a `FetchUnsupportedError()`. It would still be the fault of the developer for putting in a bad string, but a typo should be caught and be made obvious in a unit test that throws.